File: DecompiledGameScripts/cyberpunk/devices/masters/networkArea.swift

public class NetworkArea extends InteractiveMasterDevice {

  private let m_area: ref<TriggerComponent>;

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    EntityRequestComponentsInterface.RequestComponent(ri, n"area", n"TriggerComponent", false);
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_area = EntityResolveComponentsInterface.GetComponent(ri, n"area") as TriggerComponent;
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as NetworkAreaController;
  }

  private const func GetController() -> ref<NetworkAreaController> {
    return this.m_controller as NetworkAreaController;
  }

  public const func GetDevicePS() -> ref<NetworkAreaControllerPS> {
    return this.GetController().GetPS();
  }

  protected cb func OnAreaEnter(evt: ref<AreaEnteredEvent>) -> Bool {
    let entityID: EntityID = this.ExtractEntityID(evt);
    if this.IsPlayer(entityID) {
      this.GetDevicePS().AreaEntered();
    };
  }

  protected cb func OnAreaExit(evt: ref<AreaExitedEvent>) -> Bool {
    let entityID: EntityID = this.ExtractEntityID(evt);
    if this.IsPlayer(entityID) {
      this.GetDevicePS().AreaExited();
    };
  }
}

File: Definitions/Game_Definitions/classes/NetworkArea.lua
---@meta
---@diagnostic disable

---@class NetworkArea : InteractiveMasterDevice
---@field area gameStaticTriggerAreaComponent
NetworkArea = {}

---@return NetworkArea
function NetworkArea.new() return end

---@param props table
---@return NetworkArea
function NetworkArea.new(props) return end

---@param evt entAreaEnteredEvent
---@return Bool
function NetworkArea:OnAreaEnter(evt) return end

---@param evt entAreaExitedEvent
---@return Bool
function NetworkArea:OnAreaExit(evt) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function NetworkArea:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function NetworkArea:OnTakeControl(ri) return end

---@return NetworkAreaController
function NetworkArea:GetController() return end

---@return NetworkAreaControllerPS
function NetworkArea:GetDevicePS() return end

